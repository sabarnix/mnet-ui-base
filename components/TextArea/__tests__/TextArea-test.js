"use strict";

var _react = _interopRequireDefault(require("react"));

var _reactTestRenderer = _interopRequireDefault(require("react-test-renderer"));

require("jest-styled-components");

var _MnetUIBase = require("../../MnetUIBase");

var _ = require("..");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

jest.mock('react-dom');
describe('TextArea', function () {
  test('basic', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.TextArea, {
      id: "item",
      name: "item"
    })));

    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('placeholder', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.TextArea, {
      id: "item",
      name: "item",
      placeholder: "placeholder"
    })));

    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('plain', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.TextArea, {
      id: "item",
      name: "item",
      plain: true
    })));

    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('disabled', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.TextArea, {
      disabled: true,
      id: "item",
      name: "item",
      plain: true
    })));

    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('focusIndicator', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.TextArea, {
      id: "item",
      name: "item",
      focusIndicator: true
    })));

    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('fill', function () {
    var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.TextArea, {
      id: "item",
      name: "item",
      fill: true
    })));

    var tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  [true, false, 'horizontal', 'vertical'].forEach(function (resize) {
    test("resize " + resize, function () {
      var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.TextArea, {
        id: "item",
        name: "item",
        resize: resize
      })));

      var tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  ['small', 'medium', 'large'].forEach(function (size) {
    test("size " + size, function () {
      var component = _reactTestRenderer["default"].create( /*#__PURE__*/_react["default"].createElement(_MnetUIBase.MnetUIBase, null, /*#__PURE__*/_react["default"].createElement(_.TextArea, {
        id: "item",
        name: "item",
        size: size
      })));

      var tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});