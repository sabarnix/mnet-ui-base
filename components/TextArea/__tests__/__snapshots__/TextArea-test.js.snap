// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea basic 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea disabled 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
  outline: none;
  border: none;
  width: 100%;
  -webkit-appearance: none;
  opacity: 0.3;
  cursor: default;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    disabled={true}
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea fill 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
  height: 100%;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea focusIndicator 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea placeholder 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    placeholder="placeholder"
  />
</div>
`;

exports[`TextArea plain 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
  outline: none;
  border: none;
  width: 100%;
  -webkit-appearance: none;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea resize false 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
  resize: none;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea resize horizontal 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
  resize: horizontal;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea resize true 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
  resize: both;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea resize vertical 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
  resize: vertical;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`TextArea size large 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
  font-size: 14.666666666666666px;
  line-height: 18.666666666666668px;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    size="large"
  />
</div>
`;

exports[`TextArea size medium 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
  font-size: 12px;
  line-height: 16px;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    size="medium"
  />
</div>
`;

exports[`TextArea size small 1`] = `
.c0 {
  font-size: 12px;
  line-height: 16px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  box-sizing: border-box;
  font-size: inherit;
  font-family: inherit;
  border: none;
  -webkit-appearance: none;
  background: transparent;
  color: inherit;
  box-shadow: inset 1px 1px 1px 0 #cdd3e3;
  padding: 7px;
  font-weight: 600;
  margin: 0;
  border: 1px solid #cdd3e3;
  border-radius: 4px;
  width: 100%;
  font-size: 9.333333333333334px;
  line-height: 13.333333333333334px;
}

.c1::-webkit-search-decoration {
  -webkit-appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-placeholder {
  color: #e0e4ee;
}

.c1:-ms-input-placeholder {
  color: #e0e4ee;
}

.c1::-moz-focus-inner {
  border: none;
  outline: none;
}

<div
  className="c0"
>
  <textarea
    className="c1"
    id="item"
    name="item"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    size="small"
  />
</div>
`;
